### PR TITLE
Refactor PeopleList for both courses and groups

### PR DESCRIPTION
**Pull request by shichongrui**
affects: Student
refs: MBL-11665
release note: none

### CODE REVIEW

**Comment by instructure-cx** [comment]
Scan submitted to Checkmarx

File: Student/Student/People/PeopleListCell.swift [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+            return
+        }
+
+        if let avatarURL = searchRecipient.avatarURL {
+            avatar.setImageWith(avatarURL)
+            avatar.roundCorners(corners: .allCorners, radius: avatar.frame.width / 2)
+        }
+        name.text = searchRecipient.fullName
+        roles.text = searchRecipient.roles
+        roles.isHidden = searchRecipient.roles == ""
```
Comment (bluwave) [inline Student/Student/People/PeopleListCell.swift]: perhaps `searchRecipient.roles.isEmpty`?

Comment (shichongrui) [reply Student/Student/People/PeopleListCell.swift]: ah yes much better

**Comment by instructure-cx** [comment]
### Checkmarx scan completed
[Full Scan Details](https://cxmgr.inssec.info/CxWebClient/ViewerMain.aspx?scanid=1005870&projectid=17)
#### Checkmarx Scan Summary
Severity|Count
---|---
High|0
Medium|0
Low|0
Informational|0

#### Violation Summary
Severity|Count
---|---

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  This pull request will not generate a release note.
  </td>
    </tr>
  </tbody>
</table>


#### Affected Apps: Student

[MBL-11665](https://instructure.atlassian.net/browse/MBL-11665)

Coverage | New % | Master % | Delta
-------- | ----- | -------- | -----
**Canvas iOS** | 93.19% | 93.07% | 0.12%
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Pull request merged**